src/relay/backend/build_module.cc

```diff
@@ -473,23 +492,35 @@ class RelayBuildModule : public runtime::ModuleNode {
 
     // Relay IRModule -> IRModule optimizations.
     relay_module = Optimize(relay_module, targets_, params);
+
     // Get the updated function.
     auto func = Downcast<Function>(relay_module->Lookup("main"));
 
     // Generate code for the updated function.
-    graph_codegen_ = std::unique_ptr<GraphCodegen>(new GraphCodegen());
-    graph_codegen_->Init(nullptr, targets_);
-    graph_codegen_->Codegen(func);
+    const String executor_str =
+        target_host->GetAttr<String>("executor").value_or(kTvmExecutorGraph);
+    if (executor_str == kTvmExecutorGraph) {
+      executor_codegen_ = std::unique_ptr<ExecutorCodegen>(new GraphCodegen());
+    } else {
+      executor_codegen_ = std::unique_ptr<ExecutorCodegen>(new AOTCodegen());
+    }
 
-    ret_.graph_json = graph_codegen_->GetJSON();
-    ret_.params = graph_codegen_->GetParams();
+    executor_codegen_->Init(nullptr, targets_);
+    executor_codegen_->Codegen(func);
+
+    if (executor_str == kTvmExecutorGraph) {
+      ret_.graph_json = reinterpret_cast<GraphCodegen*>(executor_codegen_.get())->GetJSON();
```

Comment:
nit : I think we can have a virtual method ExecutorCodegen::UpdateOutput(const BuildOutput& ret) that implements the update as required by specialized classes.